clients.asciidoc

```diff
@@ -1,32 +1,33 @@
 
 Ethereum is defined by a formal specification called the "Yellow Paper".
 This is in contrast to, for example, Bitcoin, which is not defined in any formal way. Where Bitcoin's "specification" is the reference implementation Bitcoin Core, Ethereum's specification is documented in a paper combining an English and a mathematical (formal) specification.
-This formal specification, in addition to various Ethereum Improvement Proposals, defines the standard behavior of an Ethereum client.
+This formal specification, in addition to various Ethereum Improvement Proposals, defines the standard behavior of an Ethereum client. As major changes are made to Ethereum, the Yellow Paper is updated periodically to reflect these.
 
 As a result of having a clear formal specification, there are a number of independently developed, yet interoperable, software implementations of an Ethereum client. Ethereum has a greater diversity of implementations running on the network than any other blockchain.
 
 The vast majority of the Ethereum network runs one of the two dominant implementations: Geth and Parity. Geth is an implementation of Ethereum written in the Go programming language. Parity is a competing implementation written in Rust.
 
 In this section, we will look at the two most common clients, Geth and Parity. We'll learn to set up a node using each of them and explore some of their command-line and application programming interfaces (APIs).
 
-== Should I run a full node?
+=== Should I run a full node?
 
 The health, resilience and censorship resistance of Ethereum depends on having many independently operated and geographically dispersed full nodes. Each full node can help other new nodes obtain the block data to bootstrap their operation, as well as offer the operator an authoritative and independent verification of all transactions and contracts.
 
 However, to run a full node you will incur a significant cost in hardware resources and bandwidth. A full node must download more than 50GB of data (as of Feb 2018) and store it on a local hard drive. This data burden increases quite rapidly every day as new transactions and blocks are added. More on this topic in <<requirements>>.
 
-A full node in the _mainnet_ is not necessary for Ethereum development. You can do everything you need to do with a _testnet_ node (which stores a copy of the smaller public test blockchain), or with a local private blockchain.
+A full node running on the Ethereum Main Network (mainnet) is not necessary for Ethereum development. You can do everything you need to do with a _testnet_ node (which stores a copy of the smaller public test blockchain), or with a local private blockchain. You can also develop Ethereum applications with a local private blockchain (see <<ganache>>), or use a cloud-based Ethereum client offered by a service provider (see <<infura>>).
```

Comment:
Taking into account that the **state** (e.g.  a certain state of a/some contract(s) or the previous existence of certain libraries) of the system could be crucial to perform certain tests, it would be possible to find the case in which **the state** with which to perform a set of tests **did not exist in the Testnet** network or to reach that state would require a considerable effort.